src/main/java/bio/terra/service/filedata/azure/util/BlobContainerClientFactory.java

```diff
@@ -169,9 +174,44 @@ private String createReadOnlySasUrlForBlobUsingClient(String blobName) {
         "%s/%s?%s", blobContainerClient.getBlobContainerUrl(), blobClient.getBlobName(), sasToken);
   }
 
-  private BlobServiceSasSignatureValues createSasSignatureValues() {
+  public enum SASPermission {
+    READ_ONLY,
+    WRITE_PARQUET
+  }
+
+  private BlobServiceSasSignatureValues createSasSignatureValues(SASPermission permissionType) {
+    switch (permissionType) {
+      case WRITE_PARQUET:
+        return createSasSignatureValues(true, true, true, true, true, true, true, true, true);
```

Comment:
i think we'll also need delete.  I like this idea of prebaked sets of permissions but perhaps it should be a high-up class. This class is intended to be fairly lower level (e.g. this could get pulled into a terra-wide repo)

Reply:
Ok - You mentioned we have some changes coming to the factory soon. So, I moved any synapse specific logic out of the factory. I also changed the permission definition to something pretty basic that should be easy to change according to Jesus' changes: I'm now passing around a permission string (e.g. "r" or "rw") instead of the enum.

Reply:
(This is no longer relevant - it was handled by Jesus' PR)